server/sonar-server/src/main/java/org/sonar/server/computation/formula/CoreFormulaRepositoryImpl.java

```diff
@@ -0,0 +1,123 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.formula;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+
+import static org.sonar.api.measures.CoreMetrics.CLASSES_KEY;
+import static org.sonar.api.measures.CoreMetrics.CLASS_COMPLEXITY_KEY;
+import static org.sonar.api.measures.CoreMetrics.COMMENT_LINES_KEY;
+import static org.sonar.api.measures.CoreMetrics.COMPLEXITY_IN_CLASSES_KEY;
+import static org.sonar.api.measures.CoreMetrics.COMPLEXITY_IN_FUNCTIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.COMPLEXITY_KEY;
+import static org.sonar.api.measures.CoreMetrics.FILES_KEY;
+import static org.sonar.api.measures.CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION_KEY;
+import static org.sonar.api.measures.CoreMetrics.FILE_COMPLEXITY_KEY;
+import static org.sonar.api.measures.CoreMetrics.FUNCTIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.FUNCTION_COMPLEXITY_DISTRIBUTION_KEY;
+import static org.sonar.api.measures.CoreMetrics.FUNCTION_COMPLEXITY_KEY;
+import static org.sonar.api.measures.CoreMetrics.GENERATED_LINES_KEY;
+import static org.sonar.api.measures.CoreMetrics.GENERATED_NCLOC_KEY;
+import static org.sonar.api.measures.CoreMetrics.PUBLIC_API_KEY;
+import static org.sonar.api.measures.CoreMetrics.PUBLIC_UNDOCUMENTED_API_KEY;
+import static org.sonar.api.measures.CoreMetrics.STATEMENTS_KEY;
+
+public class CoreFormulaRepositoryImpl implements FormulaRepository {
+
+  private static final List<Formula> FORMULAS;
+
+  static {
+    ImmutableList.Builder<Formula> formulaList = ImmutableList.builder();
+    formulaList.add(
+      // Sum formulas
+      // new SumFormula(LINES_KEY),
+      new SumFormula(GENERATED_LINES_KEY),
+      // new SumFormula(NCLOC_KEY),
+      new SumFormula(GENERATED_NCLOC_KEY),
+      new SumFormula(CLASSES_KEY),
+      new SumFormula(FUNCTIONS_KEY),
+      new SumFormula(STATEMENTS_KEY),
+      new SumFormula(PUBLIC_API_KEY),
+      new SumFormula(COMMENT_LINES_KEY),
+      new SumFormula(PUBLIC_UNDOCUMENTED_API_KEY),
+      new SumFormula(COMPLEXITY_KEY),
+      new SumFormula(COMPLEXITY_IN_CLASSES_KEY),
+      new SumFormula(COMPLEXITY_IN_FUNCTIONS_KEY),
+      // new SumFormula(LINES_TO_COVER_KEY),
+      // new SumFormula(NEW_LINES_TO_COVER_KEY),
+      // new SumFormula(UNCOVERED_LINES_KEY),
+      // new SumFormula(NEW_UNCOVERED_LINES_KEY),
+      // new SumFormula(CONDITIONS_TO_COVER_KEY),
+      // new SumFormula(NEW_CONDITIONS_TO_COVER_KEY),
+      // new SumFormula(UNCOVERED_CONDITIONS_KEY),
+      // new SumFormula(NEW_UNCOVERED_CONDITIONS_KEY),
+      // new SumFormula(IT_LINES_TO_COVER_KEY),
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)